packages/nextra-theme-docs/src/sidebar.tsx

```diff
@@ -90,18 +90,54 @@ function FolderImpl({ item, anchors }: FolderProps) {
     </a>
   )
 
+  if (item.type === 'menu') {
+    const menu = item as MenuItem
+    const routes = Object.fromEntries(
+      (menu.children || []).map(route => [route.name, route])
+    )
+    const directories = Object.entries(menu.items || {}).map(([key, item]) => {
+      const route = routes[key] || {
+        name: key,
+        locale: menu.locale,
+        route: menu.route + '/' + key
+      }
+      return {
+        ...route,
+        ...item
+      }
+    })
+
+    return (
+      <li className={cn({ open, active })}>
+        {link}
+        <Collapse open={open}>
+          <Menu
+            submenu
+            directories={directories}
+            base={item.route}
+            anchors={anchors}
+          />
+        </Collapse>
+      </li>
+    )
+  }
+
   return (
     <li className={cn({ open, active })}>
-      {item.withIndexPage ? <Link href={item.route}>{link}</Link> : link}
+      {(item as Item).withIndexPage ? (
+        <Link href={item.route}>{link}</Link>
+      ) : (
+        link
+      )}
       <Collapse open={open}>
-        {Array.isArray(item.children) && (
+        {Array.isArray(item.children) ? (
```

Comment:
I remember your suggestion about either ternary or `&&` in jsx, but previous case was save since `Array.isArray` returns boolean

Reply:
Yeah in that case they are the same, it's just personal habit